main/middleware/apisix_user_test.py

```diff
@@ -62,6 +62,45 @@ def test_get_request(mocker, mock_login):
     assert user.global_id == apisix_user_info["sub"]
 
 
+@pytest.mark.django_db(transaction=True)
+def test_get_request_existing_user_no_globalid(mocker, mock_login):
+    """Test that a valid request creates a new user."""
+    close_old_connections()
+    user = UserFactory.create(email=apisix_user_info["email"], global_id=None)
+    mock_request = mocker.Mock(
+        META={
+            "HTTP_X_USERINFO": b64encode(json.dumps(apisix_user_info).encode()),
+        },
+        user=AnonymousUser(),
+    )
+    apisix_middleware = ApisixUserMiddleware(mocker.Mock())
+    apisix_middleware.process_request(mock_request)
+    mock_login.assert_called_once()
+    user.refresh_from_db()
+    assert user.username == user.username
```

Comment:
This and other `assert user.FIELD == user.FIELD` assertions will always be true, I think we want to compare these values against what's in `apisix_user_info`